src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
-        this.state.lastInitiative.actName = ConfirmValueAct.name;
-        resultBuilder.addAct(
-            new ConfirmValueAct(this, {
-                value: this.state.value,
-                renderedValue:
-                    this.state.value !== undefined ? this.props.valueRenderer(this.state.value, input) : '',
-            }),
+    ): Promise<void> {
+        const validationResult: true | ValidationFailure = await evaluateValidationProp(
+            this.props.validation,
+            this.state,
+            input,
         );
-    }
-
-    private isBareYesConfirmingValue(input: ControlInput): boolean {
-        try {
-            okIf(InputUtil.isBareYes(input));
-            okIf(this.state.lastInitiative.actName === ConfirmValueAct.name);
-            this.handleFunc = this.handleFeedbackYesAndValueNotChangedOrUndefinedWhenConfirmingValue;
-            return true;
-        } catch (e) {
-            return falseIfGuardFailed(e);
-        }
-    }
-
-    private isFeedbackYesAndValueNotChangedWhenConfirmingValue(input: ControlInput): boolean {
-        try {
-            okIf(InputUtil.isValueControlIntent(input, AmazonBuiltInSlotType.NUMBER));
-            const { feedback, action, target, values } = unpackValueControlIntent(
-                (input.request as IntentRequest).intent,
+        if (validationResult === true) {
+            this.state.confirmed = true;
+            this.state.lastInitiative.actName = undefined;
+            resultBuilder.addAct(
+                new ValueConfirmedAct(this, {
+                    value: this.state.value,
+                    renderedValue: this.props.valueRenderer(this.state.value, input),
+                }),
             );
-            const valueStr = values[0].slotValue;
-            okIf(
-                InputUtil.actionIsMatchOrUndefined(action, [
-                    ...this.props.interactionModel.actions.set,
-                    ...this.props.interactionModel.actions.change,
-                ]),
+        } else {
+            resultBuilder.addAct(
+                new InvalidValueAct<number>(this, {
+                    value: this.state.value,
+                    renderedValue: this.props.valueRenderer(this.state.value!, input),
+                    reasonCode: validationResult.reasonCode,
+                    renderedReason: validationResult.renderedReason,
+                }),
             );
-            okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
-            okIf(InputUtil.feedbackIsTrue(feedback));
-            okIf(this.state.lastInitiative.actName === ConfirmValueAct.name);
-            okIf(this.state.value === Number.parseInt(valueStr, 10));
-            this.handleFunc = this.handleFeedbackYesAndValueNotChangedOrUndefinedWhenConfirmingValue;
-            return true;
-        } catch (e) {
-            return falseIfGuardFailed(e);
+            this.askElicitationQuestion(input, resultBuilder, $.Action.Set);
```

Comment:
this one is not critical I think and should be deferred to initiative phase.  in 99% of cases it will behave identically but there will be some things that go better.

I think the following is a good dialog.. i.e. there are some questions that are high-priority and using initiative phase allows this to be encoded.
```
A: do you agree to the licence?
U: my age is seven hundred //assume this is invalid
A: Got it, 700. Do you agree to the licence?
U: yes
A: Great. Seven hundred isn't a valid age....
```

Reply:
Discussed offline, not deferring to initiative phase always since it causes conflicts on standardInitiative built-in handlers as mentioned in previous comment.